### PR TITLE
Added allEntries cli option to indicate no date filtering of data

### PR DESCRIPTION
**Pull request by danlee1025**
- Added `-a --all-entries` to indicate filtering of the data by date is not needed.
    - ~~Any suggestions on the naming of this flag? `-d` was already taken by debug. When I made it `--no-date-flag`, commander defaults the flag to `true` which can be handled but could be confusing in the code to read.~~
- Decided not to log the run when it was run with this flag since we have no `fromDate`. If we still want this run logged let me know and I can take that change out.

Testing:
When `allEntries` is provided, a `null` `fromDate` is passed to the client which is then passed down to the extractors. The `CSVModule` already has logic for handling a `null` `fromDate` so no changes were needed there. 
Follow the steps below to confirm the `allEntries` is working:

- Clear your `run-logs.json` and make sure it is an empty array.
- Run the client with this command `node src/cli.js -p src/csv.config.json -f 2020-01-01`
    - This should extract all the resources in our csvs.
- Run the client again without a `fromDate` `node src/cli.js -p src/csv.config.json`
    - No disease status nor treatment plan change(the two resources that have date filtering applied) should be extracted
- Run the client again with the `-n` flag `node src/cli.js -p src/csv.config.json -a`
    - Both disease status and treatment plan change should get extracted, ignoring the date filter.

### CODE REVIEW

**Comment by jafeltra** [comment]
I ran the steps you provided and everything was extracted as expected. I agree with you that we shouldn't log the run when using this new flag.

For the name of the flag, I had no idea that commander added special logic for flags that start with `no-`, so I learned that. I agree that it is a little odd to have the short `-n` not match the long name of the flag `--date-flag`. The only alternative name for the flag that I've thought of is potentially `--all-entries` or `--extract-all-entries`?

The code looks good to me, but I think we probably should include documentation in the README for using this flag. Do you agree?

**Comment by danlee1025** [comment]
Yea right now it already defaults to false, I kind of like `--extract-all-entries`. Oh yea I meant to add to the `README` after putting up the PR.

**Comment by jafeltra** [comment]
Oh yeah you're so right that it already defaults to false. I got sidetracked by the whole mysterious `no-` thing and lost track of what you were already doing.

**Review by julianxcarter — APPROVED** [review]
This looks good to me! Runs as expected, nice work!

File: README.md [hunk 1]
```diff
@@ -58,7 +58,7 @@ Users can specify a different location for the file by using the `--path-to-run-
...
+If a `from-date` is provided as an option when running the ICARE Extraction Client, it will be used to filter out any data elements that are recorded before that date. If a `to-date` is provided as an option, it will be used to filter out any data elements that are recorded after that date. If no `to-date` is provided, the default is today. If no `from-date` is provided, the ICARE Extraction Client will look to the run log file to find the most recent run and use the `to-date` of that run as the `from-date` for the current run, allowing users to only run the extraction on data elements that were not included in previous runs. If there are no previous run times logged, a `from-date` needs to be provided when running the extraction, unless the `all-entries` option is provided. If the `all-entries` option is provided, the `from-date` and `to-date` options will be ignored and none of the data elements will be filtered by date.
```
Comment (jafeltra) [inline README.md:61]: Do you think we should also mention here that when you use the `all-entries` option, no dates will be logged in the run log file?

Comment (danlee1025) [reply README.md]: Good catch, updated.

**Review by jafeltra — APPROVED** [review]
Looks good to me and works as expected!

**Pull request merged**